analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/LocksReleasedAllPathsBase.cs

```diff
@@ -179,12 +188,51 @@ private ProgramState RemoveLock(SymbolicContext context, ISymbol symbol)
         private static ProgramState AddLock(SymbolicContext context, IOperation operation) =>
             context.State.SetOperationConstraint(operation, context.SymbolicValueCounter, LockConstraint.Held);
 
-        private static ISymbol FirstArgumentSymbol(IInvocationOperationWrapper invocation) =>
-            IArgumentOperationWrapper.FromOperation(invocation.Arguments.First()).Value.TrackedSymbol();
+        private static ISymbol ArgumentSymbol(IInvocationOperationWrapper invocation, int argumentIndex, string parameterName) =>
+            invocation.Arguments[argumentIndex].ToArgument() is var argument
+            && argument.Parameter.Name == parameterName
+                ? argument.Value.TrackedSymbol()
+                : invocation.Arguments.SingleOrDefault(x => x.ToArgument() is var namedArgument && namedArgument.Parameter.Name == parameterName).ToArgument().Value.TrackedSymbol();
+
+        private static RefParamContext FindRefParam(SymbolicContext context) =>
+            BoolRefParamFromArgument(context, KnownType.System_Threading_Monitor, "obj", "Enter", "TryEnter")
+            ?? BoolRefParamFromArgument(context, KnownType.System_Threading_Monitor, "obj", "TryEnter")
```

Comment:
Why is there `TryEnter` on both lines?

Reply:
After the removal of some parameters from BoolRefParamFromArgument I missed to delete the redundant lines.